design/controlling-agent-updates.md

```diff
@@ -0,0 +1,85 @@
+# Controlling agent updates
+
+The agent is considered an extension of the service.
+As such, we have an auto-update mechanism to ensure that new features can rely on the agent support they require.
+Conceptually, it's no different from binaries on the app tier or JavaScript served to the web frontend.
+
+For instance, tasks express a minimum agent version required.
+If a selected agent doesn't meet that requirement, the agent is first asked to update itself.
+Other ways an agent can be updated include using a feature which demands an evergreen agent (YAML pipelines, for instance) or clicking the "upgrade all agents" button in a pool.
+
+For some highly-regulated customers, this kind of evergreen infrastructure is at odds with their security/compliance needs.
+- For instance, Customer "K" runs the agent inside a highly change-controlled datacenter.
+Every alteration must be approved by a high-ranking official in their organization after manual source inspection.
+Today, they diff between the release tags of their current version and the version being installed, looking for potential security issues or malicious changes.
+- Customer "N" isn't as concerned about the security aspect but is in a regulated space.
+They must satisfy their auditors that they're aware of all changes taking place inside their datacenter.
+
+Such customers today are doing things like ACLing the agent's installation directory so it can't update itself or recompiling the agent without update support.
```

Comment:
Maybe add a note that when this happens builds queue up because no Agents can meet their min Agent demands.

Reply:
Basically, we just do not have an experience to draw something when this process has failed.